src/hub.js

```diff
@@ -117,79 +106,122 @@ export function hub(chrome) {
 			);
 		} else {
```

Comment:
don't forget to pull `dsp/instances`, I have made some changes to `pingApp()`.

It checks if the `PING` message can be sent, and if that fails because there's not `Window`, it assumes the `app` has timed out and won't try to `PING` again, just forgets it.